apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js

```diff
@@ -1,272 +1,282 @@
+  FormContainer,
+  PageContainer,
+  BannerContainer,
+  FormFields,
+  FormFooter,
+  ProfileContainer,
+  UploadButton,
+  ProfileHeader,
+  ProfilePicture,
+  UploadedProfilePicture,
+} from './StyledOnboardingLandingPage';
 
 const initialFormValue = {
-    firstName: '',
-    firstNameTouched: false,
-    lastName: '',
-    lastNameTouched: false,
-    location: '',
-    role: '',
-    headline: '',
-    termsAndConditions: false,
-    emailMarketing: false
-}
+  firstName: '',
+  firstNameTouched: false,
+  lastName: '',
+  lastNameTouched: false,
+  location: '',
+  role: '',
+  headline: '',
+  termsAndConditions: false,
+  emailMarketing: false,
+};
 
 const initialFormValidation = {
-    isFirstNameValid: /^[a-z ,.'-]+$/i.test(initialFormValue.firstName),
-    isLastNameValid: /^[a-z ,.'-]+$/i.test(initialFormValue.lastName)
-}
+  isFirstNameValid: /^[a-z ,.'-]+$/i.test(initialFormValue.firstName),
+  isLastNameValid: /^[a-z ,.'-]+$/i.test(initialFormValue.lastName),
+};
 
 export default function OnboardingLandingPage() {
-    const [person, setPerson] = useState(initialFormValue);
-    const [formValidation, setFormValidation] = useState(validate(initialFormValue));
-    const router = useRouter()
-    let isFormValid = false;
-    useEffect(() => {
-        setFormValidation(validate(person));
-    }, [person]);
+  const { userData } = useUserDataContext();
+  const [person, setPerson] = useState({ ...initialFormValue });
+  const [formValidation, setFormValidation] = useState(
+    validate(initialFormValue)
+  );
 
-    function validate(formValue) {
-        return {
-            isFirstNameValid: formValue.firstNameTouched && /^[a-z ,.'-]+$/i.test(formValue.firstName),
-            isLastNameValid: formValue.lastNameTouched && /^[a-z ,.'-]+$/i.test(formValue.lastName),
-            isFormValid: function () { return formValue.termsAndConditions && this.isFirstNameValid && this.isLastNameValid }
-        }
-    }
+  const [profilePicture, setProfilePicture] = useState(
+    userData?.profile?.profilePictureUrl
+  );
 
-    function onFirstNameChange(e) {
-        console.log(person);
-        setPerson({ ...person, firstName: e.target.value, firstNameTouched: true });
-        console.log("VALIDATION: ", formValidation)
-    }
+  const [uploadedProfilePicture, setUploadedProfilePicture] = useState(null);
 
-    function onLastNameChange(e) {
-        console.log(person);
-        setPerson({ ...person, lastName: e.target.value, lastNameTouched: true });
-        console.log("VALIDATION: ", formValidation)
-    }
+  const router = useRouter();
 
-    function onLocationChange(e) {
-        console.log(person);
-        setPerson({ ...person, location: e.target.value })
-    }
+  useEffect(() => {
+    setFormValidation(validate(person));
+  }, [person]);
 
-    function onRoleChange(e) {
-        setPerson({ ...person, role: e.target.value });
-        console.log(person);
-    }
+  function validate(formValue) {
```

Comment:
How would we tell the user what failed to validate?

Reply:
We can add an error check that can output an alert?

Reply:
That might be easier for user to understand what was invalid. @judeamos do you have an opinion on this?

Reply:
What about this as a possible solution: 
```JavaScript
function validate(formValue) {
  const errors = {};
  if (!formValue.firstNameTouched) {
    errors.isFirstNameValid = false;
    errors.firstNameError = 'First name is required';
  } else if (!/^[a-z ,.'-]+$/i.test(formValue.firstName)) {
    errors.isFirstNameValid = false;
    errors.firstNameError = 'First name is not valid';
  } else {
    errors.isFirstNameValid = true;
  }

  if (!formValue.lastNameTouched) {
    errors.isLastNameValid = false;
    errors.lastNameError = 'Last name is required';
  } else if (!/^[a-z ,.'-]+$/i.test(formValue.lastName)) {
    errors.isLastNameValid = false;
    errors.lastNameError = 'Last name is not valid';
  } else {
    errors.isLastNameValid = true;
  }

  errors.isFormValid = () => formValue.termsAndConditions && errors.isFirstNameValid && errors.isLastNameValid;

  return errors;
}
```

```JavaScript
<InputField
  error={formValidation.firstNameError}
  label="First Name"
  name="First Name"
  onChange={onFirstNameChange}
  placeholder="John"
  touched={person.firstNameTouched && !formValidation.isFirstNameValid}
  required
/>
<InputField
  error={formValidation.lastNameError}
  label="Last Name"
  name="Last Name"
  onChange={onLastNameChange}
  placeholder="Doe"
  touched={person.lastNameTouched && !formValidation.isLastNameValid}
  required
/>
```

Reply:
Can validate throw an error and do a try catch in the caller? If not, I think your solution works except errors should be a var not const.

Reply:
Yes that worked here:

```JavaScript
function validate(formValue) {
  var errors = {};
  if (!formValue.firstNameTouched) {
    errors.isFirstNameValid = false;
    errors.firstNameError = 'First name is required';
  } else if (!/^[a-z ,.'-]+$/i.test(formValue.firstName)) {
    errors.isFirstNameValid = false;
    errors.firstNameError = 'First name is not valid';
  } else {
    errors.isFirstNameValid = true;
  }

  if (!formValue.lastNameTouched) {
    errors.isLastNameValid = false;
    errors.lastNameError = 'Last name is required';
  } else if (!/^[a-z ,.'-]+$/i.test(formValue.lastName)) {
    errors.isLastNameValid = false;
    errors.lastNameError = 'Last name is not valid';
  } else {
    errors.isLastNameValid = true;
  }

  errors.isFormValid = () => formValue.termsAndConditions && errors.isFirstNameValid && errors.isLastNameValid;

  return errors;
}

```

```JavaScript
<InputField
  error={formValidation.firstNameError}
  label="First Name"
  name="First Name"
  onChange={onFirstNameChange}
  placeholder="John"
  touched={person.firstNameTouched && !formValidation.isFirstNameValid}
  required
/>
<InputField
  error={formValidation.lastNameError}
  label="Last Name"
  name="Last Name"
  onChange={onLastNameChange}
  placeholder="Doe"
  touched={person.lastNameTouched && !formValidation.isLastNameValid}
  required
/>

```